lib/user_interface.rb

```diff
@@ -70,8 +58,24 @@ def display_contact(contact)
     end
   end
 
+  def add_another_contact?
+    output.print ANOTHER_CONTACT_PROMPT
+    collect_vaild_input { |user_input| user_input.match?(/[yn]/i) }.match?(/y/i)
```

Comment:
You could replace `user_input.match?(/[yn]/i)` with a boolean method `valid_yes_no_answer?` to follow the same pattern of the others.

Also, I would put the `y` in a constant too. This is because what the user types here is related to what you tell them to reply in the prompt. For Spanish, this would be `s/n`, so answering `y` would fail.

I also wouldn't do `match?(/y/i)` at the end. It's simpler and more intention revealing to just do `== 'y'`, because in this case you want strict equality.